### PR TITLE
add blog post about custom sd implementation

### PR DESCRIPTION
**Pull request by cstyan**
moving from the google doc to here, I've made a few more grammatical changes plus added more links and code blocks

cc @brian-brazil

### CODE REVIEW

File: content/blog/2018-06-27-implementing-custom-sd.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+## Implementing Custom Service Discovery
+
+Prometheus contains built in integrations for many service discovery (SD) systems such as Consul,
+Kubernetes, and public cloud providers such as Azure. However, we can’t provide integration
+implementations for every service discovery option out there. The Prometheus team is already stretched
+thin supporting the current set of SD integrations, so maintaining an integration for every possible SD
+option isn’t feasible. In many cases the current SD implementations have been contributed by people
+outside the team and then not maintained or tested well. We want to commit to only providing direct
+integration with service discovery mechanisms that we know we can maintain, and that work as intended.
+For this reason, there is currently a moratorium on new SD integrations
```
Comment (brian-brazil) [inline content/blog/2018-06-27-implementing-custom-sd.md]: full stop

File: content/blog/2018-06-27-implementing-custom-sd.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+However, we know there is still a desire to be able to integrate with other SD mechanisms, such as 
+Docker Swarm. Recently a small code change plus an example was committed to the documentation directory
```
Comment (brian-brazil) [inline content/blog/2018-06-27-implementing-custom-sd.md]: link to the directory?

File: content/blog/2018-06-27-implementing-custom-sd.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+within the Prometheus repository for implementing a custom service discovery integration without having
+to merge it into the main Prometheus binary. The code change allows us to make use of the internal
```
Comment (brian-brazil) [inline content/blog/2018-06-27-implementing-custom-sd.md]: We should probably make it clear here that we're not going to merge them into the docs directory either, we've had that confusion with the remote adapter example.

We should link to https://prometheus.io/docs/operating/integrations/#file-service-discovery too

**Comment by cstyan** [comment]
addressed your comments, wasn't entirely sure what we wanted to say regarding the file_sd link but I am assuming we would potentially list known stable adapter/file_sd implementations on that page

File: content/blog/2018-06-27-implementing-custom-sd.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+Discovery Manager code to write another executable that interacts with a new SD mechanism and outputs
+a file that is compatible with Prometheus' file\_sd. By co-locating Prometheus and our new executable
+we can configure Prometheus to read the file\_sd-compatible output of our executable, and therefore
+scrape targets from that service discovery mechanism. In the future this will enable us to move SD
+integrations out of the main Prometheus binary, as well as to move stable SD integrations that make
+use of the adapter into the Prometheus 
+[discovery](https://github.com/prometheus/prometheus/tree/master/discovery) package. 
+
+Popular integrations that are implemented with the adapter code may end up listed
```
Comment (brian-brazil) [inline content/blog/2018-06-27-implementing-custom-sd.md]: Popularity isn't a requirement, I'd say that we list file_sd integrations such as those using this adapter on our integrations page.

**Comment by cstyan** [comment]
@brian-brazil PTAL, I've rephrased the sentences you pointed out.

**Comment by brian-brazil** [comment]
This looks good. Can you bump the date, as it's a bit too different by now?

**Comment by cstyan** [comment]
:+1: updated the date to July 5 :)

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**